src/optimizer/filter_combiner.cpp

```diff
@@ -152,6 +153,91 @@ bool FilterCombiner::HasFilters() {
 	return has_filters;
 }
 
+
+vector<TableFilter> FilterCombiner::GenerateTableScanFilters(std::function<void(unique_ptr<Expression> filter)> callback){
+    vector<TableFilter> tableFilters;
+    //! First, we figure the filters that have constant expressions that we can push down to the table scan
+    for (auto & constant_value: constant_values) {
+		if (constant_value.second.size() > 0) {
+			for (idx_t i = 0; i < constant_value.second.size(); ++i) {
+				if ((constant_value.second[i].comparison_type == ExpressionType::COMPARE_EQUAL ||
+                    constant_value.second[i].comparison_type == ExpressionType::COMPARE_GREATERTHAN ||
+                    constant_value.second[i].comparison_type == ExpressionType::COMPARE_GREATERTHANOREQUALTO ||
+                    constant_value.second[i].comparison_type == ExpressionType::COMPARE_LESSTHAN ||
+                    constant_value.second[i].comparison_type == ExpressionType::COMPARE_LESSTHANOREQUALTO
+                    ) && (constant_value.second[i].constant.type == TypeId::INT8 ||
+                    constant_value.second[i].constant.type == TypeId::INT16 ||
+                    constant_value.second[i].constant.type == TypeId::INT32 ||
+                    constant_value.second[i].constant.type == TypeId::INT64 ||
+                    constant_value.second[i].constant.type == TypeId::DECIMAL ||
+                    constant_value.second[i].constant.type == TypeId::FLOAT ||
+                    constant_value.second[i].constant.type == TypeId::DOUBLE)) {
+                    //! Here we check if these filters are column references
+                    auto filter_exp = equivalence_map.find(constant_value.first);
+                    string rowid = "rowid";
+                    if (filter_exp->second.size() == 1 && filter_exp->second[0]->type == ExpressionType::BOUND_COLUMN_REF && filter_exp->second[0]->alias != rowid){
```

Comment:
Using the alias is a bit hacky, "rowid" can also be a column name. You should check if the used column id in the base table is "COLUMN_IDENTIFIER_ROW_ID".